lib/grakn/cache/janitor.ex

```diff
@@ -52,6 +52,7 @@ defmodule Grakn.Cache.Janitor do
       type, error ->
         error_spec = {type, error, __STACKTRACE__}
         Logger.warn("Couldn't close session #{session_id}, error: #{inspect(error_spec)}")
+        Cache.delete(key)
```

Comment:
We could remove both `Cache.delete/1`calls and use the keyword `after` to explicitly ensure that this operation is always executed. What do you think of something like that? 

```
 defp close_session(key, %{session_id: session_id, name: name}) do
    try do
      {:ok, _} = Grakn.command(name, Command.close_session(session_id))
    catch
      type, error ->
        error_spec = {type, error, __STACKTRACE__}
        Logger.warn("Couldn't close session #{session_id}, error: #{inspect(error_spec)}")
    after
      Cache.delete(key)
    end 
```

Reply:
@lorenzosinisi Yes, it possible, but for the future it is better to have it explicitly, because you not always want to delete session in a future, you want retry if grakn wasn't available:

```elixir
try do
      case Grakn.command(name, Command.close_session(session_id)) do
         ...`grakn not available` -> `do nothing, it will retry on next run`
      end
```

Make sense?

Reply:
I see, alright let's leave it like that